src/course-unit/sidebar/index.jsx

```diff
@@ -1,27 +1,51 @@
 import PropTypes from 'prop-types';
-import { useSelector } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import classNames from 'classnames';
-import { Card } from '@edx/paragon';
+import { Card, useToggle } from '@edx/paragon';
+import { InfoOutline as InfoOutlineIcon } from '@edx/paragon/icons';
+import { useIntl } from '@edx/frontend-platform/i18n';
 
+import ModalNotification from '../../generic/modal-notification';
+import { editCourseUnitVisibilityAndData } from '../data/thunk';
 import { getCourseUnitData } from '../data/selectors';
+import { PUBLISH_TYPES } from '../constants';
 import { SidebarBody, SidebarFooter, SidebarHeader } from './components';
 import useCourseUnitData from './hooks';
+import messages from './messages';
 
-const Sidebar = ({ isDisplayUnitLocation }) => {
+const Sidebar = ({ blockId, isDisplayUnitLocation }) => {
   const {
     title,
     locationId,
     releaseLabel,
     visibilityState,
     visibleToStaffOnly,
   } = useCourseUnitData(useSelector(getCourseUnitData));
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const [isOpenDiscardModal, openDiscardModal, closeDiscardModal] = useToggle(false);
+  const [isOpenVisibleModal, openVisibleModal, closeVisibleModal] = useToggle(false);
+
+  const handleCourseUnitVisibility = () => {
+    closeVisibleModal();
+    dispatch(editCourseUnitVisibilityAndData(blockId, PUBLISH_TYPES.republish, null));
+  };
+
+  const handleCourseUnitDiscardChanges = () => {
+    closeDiscardModal();
+    dispatch(editCourseUnitVisibilityAndData(blockId, PUBLISH_TYPES.discardChanges));
+  };
+
+  const handleCourseUnitPublish = () => {
+    dispatch(editCourseUnitVisibilityAndData(blockId, PUBLISH_TYPES.makePublic));
+  };
 
   return (
     <Card
+      data-testid={isDisplayUnitLocation ? 'course-unit-location-sidebar' : 'course-unit-sidebar'}
```

Comment:
We can avoid passing testing staff here by accepting `const Sidebar = ({ blockId, isDisplayUnitLocation, ...props })` and then in the test we can pass `data-testid`.

Best practice is to pass `data-testid` in the component only if there is no way to do props spreading.

Reply:
Agree, thanks 👍